Client/Frontend/UserScript/UserScript.js

```diff
@@ -0,0 +1,67 @@
+//
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ + * License, v. 2.0. If a copy of the MPL was not distributed with this
+ + * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+var noImageModeEnabled = false;
+var observers = new Array();
+var _firefox_NoImageMode = {
+    setNoImageMode: function(enabled) {
+        noImageModeEnabled = enabled;
+        for (var i=0; i<observers.length; i++) {
+            var observer = observers.shift();
+            observer.disconnect();
+        }
+        if (enabled) {
+            _firefox_NoImageMode.setStyle();
+            var bodys = document.getElementsByTagName("body");
+            for (var i=0; i<bodys.length; i++) {
+                _firefox_NoImageMode.blockImgs(bodys[i], noImageModeEnabled);
+                _firefox_NoImageMode.elemListener(bodys[i], noImageModeEnabled);
+            }
+        }
+    },
+
+    setStyle: function() {
+        var no_image_css_id = "iBrowser_no_image_css";
+        var no_image_css = '*{background-image:none !important;}img,iframe{visibility:hidden !important;}';
+        var newCss = document.getElementById(no_image_css_id);
+        if(newCss == undefined){
+            var cssStyle = document.createElement('style');
+            cssStyle.type = 'text/css';
+            cssStyle.id = no_image_css_id;
+            if (cssStyle.styleSheet) {
+                cssStyle.styleSheet.cssText = no_image_css;
+            } else {
+                cssStyle.appendChild(document.createTextNode(no_image_css));
+            }
+            document.documentElement.appendChild(cssStyle);
+        } else {
+            newCss.innerHTML = no_image_css;
+        }
+    },
+
+    blockImgs: function(elem, enabled) {
```

Comment:
Nit: naming, use `blockImages`.

The Great Vowel Shortage ended sometime ago son.